components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -219,6 +230,9 @@ void BraveProfileSyncServiceImpl::OnNudgeSyncCycle(RecordsListPtr records) {
 
 BraveProfileSyncServiceImpl::~BraveProfileSyncServiceImpl() {
   network_connection_tracker_->RemoveNetworkConnectionObserver(this);
+  if (is_model_loaded_observer_set_) {
```

Comment:
you don't have to check before removing. Removing an observer that doesn't exist is a no-op

Reply:
if this flag is still needed for testing then it should be renamed to `is_model_loaded_observer_set_for_testing`

Reply:
but it seems like it would be better to test with a mock observer instead

Reply:
thanks, I recall, I got failed DCHECK crash one time.
I would be happy to get rid of `is_model_loaded_observer_set_`

Reply:
this was fixed at https://github.com/brave/brave-core/pull/4588/files#diff-ab344835b09fb6ac1c601af787b116bbR230